flowmachine/tests/test_cache_utils.py

```diff
@@ -0,0 +1,396 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Tests for cache management utilities.
+"""
+from cachey import Scorer
+from unittest.mock import Mock
+
+import pytest
+
+from flowmachine.core import Table
+from flowmachine.core.cache import (
+    get_compute_time,
+    shrink_below_size,
+    shrink_one,
+    get_size_of_cache,
+    get_size_of_table,
+    get_score,
+    get_query_object_by_id,
+    get_cached_query_objects_ordered_by_score,
+    touch_cache,
+    get_max_size_of_cache,
+    set_max_size_of_cache,
+    get_cache_half_life,
+    set_cache_half_life,
+    invalidate_cache_by_id,
+    cache_table_exists,
+)
+from flowmachine.features import daily_location
+
+
+def test_scoring(flowmachine_connect):
+    """
+    Test that score updating algorithm is correct by comparing to cachey as reference implementation
+    """
+    dl = daily_location("2016-01-01").store().result()
+    dl_time = get_compute_time(flowmachine_connect, dl.md5)
+    dl_size = get_size_of_table(flowmachine_connect, *dl.table_name.split(".")[::-1])
+    initial_score = get_score(flowmachine_connect, dl.md5)
+    cachey_scorer = Scorer(1000)
+    cache_score = cachey_scorer.touch("dl", dl_time / dl_size)
+    assert cache_score == pytest.approx(initial_score)
+
+    # Touch again
+    new_score = touch_cache(flowmachine_connect, dl.md5)
+    updated_cache_score = cachey_scorer.touch("dl")
+    assert updated_cache_score == pytest.approx(new_score)
+
+    # Add another unrelated cache record, which should have a higher initial score
+    dl_2 = daily_location("2016-01-02").store().result()
+    dl_time = get_compute_time(flowmachine_connect, dl_2.md5)
+    dl_size = get_size_of_table(flowmachine_connect, *dl_2.table_name.split(".")[::-1])
+    cache_score = cachey_scorer.touch("dl_2", dl_time / dl_size)
+    assert cache_score == pytest.approx(get_score(flowmachine_connect, dl_2.md5))
+
+
+def test_touch_cache_record_for_query(flowmachine_connect):
+    """
+    Touching a cache record for a query should update access count, last accessed, & counter.
+    """
+    table = daily_location("2016-01-01").store().result()
+
+    assert (
+        1
+        == flowmachine_connect.fetch(
+            f"SELECT access_count FROM cache.cached WHERE query_id='{table.md5}'"
```

Comment:
It feels to me these SQL statements could also be useful helper functions? E.g. `get_access_count(table.md5)`, `get_last_access_time(table.md5)`?